changelog/0087-upgrade-to-es-6.yml

```
@@ -0,0 +1,405 @@
+
+details: |
+  Currently, CommCareHQ is configured to run with Elasticsearch 5.6.16. The support for Elasticsearch 6.8.23 has been added in HQ. So everyone who is hosting CommCareHQ should upgrade to this version.
+  The support for Elasticsearch 5.6.16 will be removed soon.
+
+update_steps: |
+  The Elasticsearch upgrade is a multi-step process.
+  1. Reindex all HQ indices.
+  2. Swap the indices to start reading and writing to the new index.
+  3. Upgrade the Elasticsearch version.
+  4. Verify the upgrade.
+
+  ## Reindex all indices
+
+  1. Deploy the latest version of CommCare HQ.
+  2. Ensure that there is enough free space available in your cluster. The command to estimate disk space required for reindexing is:
+
+      ```sh
+      cchq <env> django-manage elastic_sync_multiplexed estimated_size_for_reindex
+      ```
+  3. Check disk usage on each node
+      - If you have separate data nodes, check disk usage on data nodes
+          ```sh
+          cchq <env>  run-shell-command es_data "df -h /opt/data" -b
+          ```
+      - If you don't have separate data nodes, check disk usage on ES nodes.
+          ```sh
+          cchq <env>  run-shell-command elasticsearch "df -h /opt/data" -b
+          ```
+      This will return disk usage for each node. You can check if the cumulative available space across all nodes is greater than the total recommended space from the `estimated_size_for_reindex` output.
+
+  4. Update the `environments/<env>/public.yml`
+    - Before we start reindexing, we need to ensure that the following parameters are set to the correct values in the `public.yml` file
+
+      #####  Elasticsearch Major Version for which we will be reindexing
+      ```yaml
+      ELASTICSEARCH_MAJOR_VERSION: 6
+      ```
+
+      ##### Index Multiplexer Settings
+      ```yaml
+      ES_APPS_INDEX_MULTIPLEXED: True
+      ES_CASE_SEARCH_INDEX_MULTIPLEXED: True
+      ES_CASES_INDEX_MULTIPLEXED: True
+      ES_DOMAINS_INDEX_MULTIPLEXED: True
+      ES_FORMS_INDEX_MULTIPLEXED: True
+      ES_GROUPS_INDEX_MULTIPLEXED: True
+      ES_SMS_INDEX_MULTIPLEXED: True
+      ES_USERS_INDEX_MULTIPLEXED: True
+      ```
+      The multiplexer settings when set to `True` will start writing data to both the old and new indices. But reads will only go to the old index.
+
+      ##### Index Swap Settings
+      ```yaml
+      ES_APPS_INDEX_SWAPPED: False
+      ES_CASE_SEARCH_INDEX_SWAPPED: False
+      ES_CASES_INDEX_SWAPPED: False
+      ES_DOMAINS_INDEX_SWAPPED: False
+      ES_FORMS_INDEX_SWAPPED: False
+      ES_GROUPS_INDEX_SWAPPED: False
+      ES_SMS_INDEX_SWAPPED: False
+      ES_USERS_INDEX_SWAPPED: False
+      ```
+      The index swap settings when set to `True` will start reading and writing to the new index. We will set these to `False` for now. They will be set to `True` after the reindexing process is complete.
```

nit: I'd rearrange this just to clearly say these should be false for now.
```suggestion
      Ensure the index swapped settings are `False`. When set to `True`, reads and writes will go to the new index. They will be set to `True` after the reindexing process is complete.
```